jdaviz/app.py

```diff
@@ -1988,6 +2002,99 @@ def _get_viewer_item(self, ref_or_id):
+
+        Unlike dataset names, the attempted renaming of a subset to an existing
+        subset label will not append a number (e.g Subset 1 repeated because
+        Subset 1(1)). If the name exists, a warning will be raised and the
+        original subset name will be returned.
+
+        """
+
+        # get active selection, if there is one
+        if self.session.edit_subset_mode.edit_subset == []:
+            subset_selected = None
+        else:
+            subset_selected = self.session.edit_subset_mode.edit_subset[0].label
+
+        # remove the current selection label from the set of labels, because its ok
+        # if the new subset shares the name of the current selection (renaming to current name)
+        if subset_selected in self._reserved_labels:
+            self._reserved_labels.remove(subset_selected)
+
+        # now check `subset_name` against list of non-active current subset labels
+        # and warn and return if it is
+        if subset_name in self._reserved_labels:
+            if raise_if_invalid:
+                raise ValueError("Cannot rename subset to name of an existing subset"
+                                 f" or data item: ({subset_name}).")
+            return False
+
+        elif not subset_name.replace(" ", "").isalnum():
+            if raise_if_invalid:
+                raise ValueError("Subset labels must be purely alphanumeric")
+            return False
+
+        else:
+            split_label = subset_name.split(" ")
+            if split_label[0] == "Subset" and split_label[1].isdigit():
+                if raise_if_invalid:
+                    raise ValueError("The pattern 'Subset N' is reserved for "
+                                     "auto-generated labels")
+                return False
+
+        return True
+
+    def _rename_subset(self, old_label, new_label, subset_group=None, check_valid=True):
+        # Change the label of a subset, making sure it propagates to as many places as it can
+        # I don't think there's an easier way to get subset_group by label, it's just a tuple
+        if subset_group is None:
+            for s in self.data_collection.subset_groups:
+                if s.label == old_label:
+                    subset_group = s
+                    break
+            # If we couldn't find a matching subset group, raise an error
+            else:
+                raise ValueError(f"No subset named {old_label} to rename")
+
+        if check_valid:
+            if self._check_valid_subset_label(new_label):
+                subset_group.label = new_label
+        else:
+            subset_group.label = new_label
+
+        # Update layer icon
+        self.state.layer_icons[new_label] = self.state.layer_icons[old_label]
+        _ = self.state.layer_icons.pop(old_label)
+
+        # Updated extracted spectrum if applicable and update _update_live_plugin_results dict.
+        if self.config == 'cubeviz':
```

Comment:
does it hurt to leave this if-statement out for generalization (it'll just cost an empty for-loop for any case that doesn't use live plugin results)?

Reply:
Eh it's probably ok to take out.